### ISSUE TITLE
adding scrobbling?

### ISSUE THREAD

**Issue by ninep9up** [zemoto/WhatAmIHearing#3]
this is super cool
as mush as cool as writing an extension for shazam web punching heard audio into a database in the background
but considering no one is as experienced
could you please add scrobbling?

like you know, i've tried macos on virtualbox
and that shazam was so fluent
i was like no way

then i decided what if android emus are as cool
but no, they can't even approach a real phone running shazam

obv idea of that is to recog tracks you wouldn't normally find tagged
like listening to stations on worldwide radio or stuff
so maybe like
what if you add a listener process on windows sound mixer or stuff and like autopress every N minutes if the sound is consistently high
because like for voice it's not consistently high
and also let spec that N in settings
or check for silence (i mean if you go for histograms analysis, it's nothing big)

no i'm not writing a diploma
this is like a cool thing you did

**Comment by ninep9up** [comment]
ok real cool thing would be to detect between-track silence and then just scrobble again after track duration
people think of silence as something hard to detect but it's actually a tenth matter to a second
so whether you jinx for tenth a second between tracks or just switch a radio station
that's like
ideal

man
what am i doing here

**Comment by ninep9up** [comment]
whether i'm living in 3000s or not
please try

**Comment by ninep9up** [comment]
ok like
autopress hotkey on the silence matter based on the histogram

**Comment by ninep9up** [comment]
also, off the autopress if it's detected both silence-unproof in the sample and in the result and perhaps add a proof-histogram listener to automatically start

**Comment by ninep9up** [comment]
(don't forget scrobbling)

**Comment by ninep9up** [comment]
(if you don't want to)

**Comment by ninep9up** [comment]
man this is gauss is it
all this tenths were always gauss

**Comment by ninep9up** [comment]
before i went sleeping i scribbled on the paper
"scrobble and press on the change to silence"
so basically you just

artist = x, song = y
gauss < median { recordmap }
api { recordmap } = !silence [x,y]
scrobble (!silence [x,y])

**Comment by ninep9up** [comment]
yeah man i was googling they have name for these tenths of seconds
decibels
so basically you check for the amplitude fades

**Comment by ninep9up** [comment]
https://youlean.co/online-loudness-meter/

**Comment by ninep9up** [comment]
so if you measure second's tenth gonna fall twice in a row in silence
so anything going from 100 to 75 in a tenth is basically it

**Comment by ninep9up** [comment]
man this was easy

**Comment by ninep9up** [comment]
so like, please implement it ok
you are the best
also to be my level of abstraction you just need to think everything has sense
not bears it
this is called faith
later

**Comment by zemoto** [comment]
It's a good idea in theory. Just on a quick google I think I would have to integrate with Last.fm to do any kind of scrobbling, as Shazam doesn't have any (publically facing) logic in that area. 

The app would listen, occasionally send requests to Shazam if a criteria is met. Then if a song is found send that to Last.fm if their scrobbling criteria is met. Then you would have to go to Last.fm to get that actual scrobbling result data.

Only problem I can see is that it would likely eat up your limited monthly Shazam request quota pretty quickly. On top of the fact that this would be a rather large task, so it may take a while before you hear anything from me in terms of this feature.

**Comment by zemoto** [comment]
After multiple attempts I couldn't really find a good way to implement this in a way that... worked. Maybe someone more knowledgeable can take a shot, but closing this issue for this particular repo.
